Jest 28 upgrade

tetarchus opened the pull request:
Upgrades to Jest 28 and fixes any test/type errors that came up. 

I don't use this package directly, only through a third party library so have been unable to test in use, but all tests are successfully passing. 

Please let me know if any edits need to be made, or if anything isn't working when testing in an actual use-case.

Should fix #796

tetarchus committed:
Adds peer deps for jest 28

tetarchus committed:
Upgrades jest packages

tetarchus committed:
Upgrades @types/jest

tetarchus committed:
Fixes `runTypes` to match parent class

tetarchus committed:
Downgrade @types/jest to prevent conflict with ts-jest

tetarchus committed:
Updates JestPlaywrightProjectConfig type to account for Jest 28

tetarchus committed:
Updates PlaywrightEnvironment

- Adds default require for RootEnv
- Adds switch if config uses Jest 28 shape

mxschmitt commented on package.json:
```diff
@@ -52,14 +52,14 @@
-    "jest-circus": "^26.6.3 || ^27.0.0",
-    "jest-environment-node": "^26.6.2 || ^27.0.0",
-    "jest-runner": "^26.6.3 || ^27.0.0"
+    "jest": "^26.6.3 || ^27.0.0 || ^28.0.0",
```
does this mean after this change we are still compatible with jest 26 and 27?

mxschmitt replied:
Because if it means we are not compatible with <28 then we need to release it as a major change, otherwise just as a minor version bump.

tetarchus committed:
Updates runner and environment

- Should now work with the existing (26/27)
versions and the latest version (28)

tetarchus committed:
Removes debugging comments

HarelM commented:
Any news about this?
I would love to move forward to jest 28 and I believe this will help me so...

syntactic-salt commented:
Any plan to move forward with this upgrade?

mxschmitt commented on package.json:
```diff
@@ -52,14 +52,14 @@
-    "jest-circus": "^26.6.3 || ^27.0.0",
-    "jest-environment-node": "^26.6.2 || ^27.0.0",
-    "jest-runner": "^26.6.3 || ^27.0.0"
+    "jest": "^26.6.3 || ^27.0.0 || ^28.0.0",
```
```suggestion
    "jest": "^28.0.0",
```
